### PR TITLE
Add support to sale and get information to orders paid with boleto

### PR DESCRIPTION
**Pull request by bvicenzo**
This PR introduces changes at sales creation and sales details.
This allows us to create and search orders paid with boleto.

### CODE REVIEW

File: lib/fake_braspag/sales.rb [hunk 1]
```diff
@@ -2,7 +2,9 @@ module FakeBraspag
   class Sales < Sinatra::Base
```
Comment (pmatiello) [inline lib/fake_braspag/sales.rb]: If you like, some conditionals can be avoided below by creating a Hash such as:

```ruby
METHOD_SPECIFIC = {
  'CreditCard' => {
    views: {
      find: : get_credit_card_sale,
      authorize: :sales_authorize
    },
    input: {
      'paymentMethod' => 'CreditCard',
      'cardNumber' => @params['Payment']['CreditCard']['CardNumber'],
      'saveCard' => @params['Payment']['CreditCard']['SaveCard']
    }
  }
  'Boleto' = {
    views: {
      find: :get_boleto_sale
      authorize: sales_boleto
    },
    input: {
      'paymentMethod' => 'Boleto'
    }
  }
}
```

And then applying the modifications specified in the comments below.

File: lib/fake_braspag/sales.rb [hunk 2]
```diff
@@ -2,7 +2,9 @@ module FakeBraspag
...
     get '/:PaymentId' do
       if ResponseToggler.enabled?('get_sale')
-        jbuilder :get_sale
+        order = Order.find!(params[:PaymentId])
+        @sale = SalePresenter.new(order)
+        jbuilder order.PaymentMethod == 'CreditCard' ? :get_credit_card_sale : :get_boleto_sale
```
Comment (pmatiello) [inline lib/fake_braspag/sales.rb]:
```ruby
jbuilder METHOD_SPECIFIC[order.PaymentMethod][:views][:find]
```

File: lib/fake_braspag/sales.rb [hunk 3]
```diff
@@ -15,10 +17,10 @@ class Sales < Sinatra::Base
         order.authorize!
         @sale = SalePresenter.new(order)
 
-        jbuilder :sales_authorize
+        jbuilder order.PaymentMethod == 'CreditCard' ? :sales_authorize : :sales_boleto
```
Comment (pmatiello) [inline lib/fake_braspag/sales.rb]:
```ruby
jbuilder METHOD_SPECIFIC[order.PaymentMethod][:views][:authorize]
```

File: lib/fake_braspag/sales.rb [hunk 4]
```diff
@@ -15,10 +17,10 @@ class Sales < Sinatra::Base
...
       rescue Order::AuthorizationFailureError
         @sale = SalePresenter.new(order)
-        jbuilder :sales_authorize
+        jbuilder order.PaymentMethod == 'CreditCard' ? :sales_authorize : :sales_boleto
```
Comment (pmatiello) [inline lib/fake_braspag/sales.rb]:
```ruby
jbuilder METHOD_SPECIFIC[order.PaymentMethod][:views][:authorize]
```

**Comment by pmatiello** [comment]
:sheep:

**Comment by j133y** [comment]
:sunglasses:

**Pull request merged**